lib/streamlit/commands/execution_control.py

```diff
@@ -143,7 +141,7 @@ def rerun(  # type: ignore[misc]
                 query_string=query_string,
                 page_script_hash=page_script_hash,
                 fragment_id_queue=_new_fragment_id_queue(ctx, scope),
-                is_fragment_scoped_rerun=True,
+                is_fragment_scoped_rerun=scope == "fragment",
```

Comment:
This is a mildly embarrassing typo / oversight that doesn't really have any effect. In the incorrect case where we have an app-scoped rerun, `fragment_id_queue` and `is_fragment_scoped_rerun==True`, but we use `fragment_id_queue` to determine behavior so the incorrect flag doesn't particularly do anything.

I'll add a test for this before this PR is submitted, though.